### ISSUE TITLE
[MPS] Multilayer LSTM with dropout=1.0 returns all NaN

### ISSUE THREAD

**Issue by bogatyy** [pytorch/pytorch#190057]
### 🐛 Describe the bug

A training-mode `nn.LSTM` with `num_layers >= 2` and `dropout=1.0` returns NaN
for every output element on MPS. The hand-rolled inter-layer dropout builds an
all-zero keep mask at p=1 and then multiplies it by `1/(1-p) = inf`, so every
masked element becomes `0 * inf = NaN`. PyTorch's defined `dropout=1` behavior
is a zero inter-layer input, not NaN — CPU returns finite output.

```python
import torch

torch.manual_seed(0)
lstm = torch.nn.LSTM(4, 4, num_layers=2, dropout=1.0)
x = torch.randn(3, 2, 4)
print(lstm(x)[0].isfinite().all())          # CPU: True
print(lstm.to("mps")(x.to("mps"))[0].isnan().all())  # MPS: True (all NaN)
```

Output on a source build of current main (2.14.0a0+git2662808); also
reproduces on 2.13.0 stable and nightly 2.14.0.dev20260710.

The scale constant is built unconditionally as `1.0 / (1.0 - dropout_p)`:
https://github.com/pytorch/pytorch/blob/26628087023e5102381849e9a27c3f09b6d485a4/aten/src/ATen/native/mps/operations/RnnOps.mm#L249-L252

### Versions

PyTorch 2.14.0a0+git2662808 (source build of main), 2.13.0, and nightly 2.14.0.dev20260710; macOS 26.5.2 (arm64), Apple M5 Max, Python 3.12.13.

cc @mikaylagawarecki @kulinseth @malfet @DenisVieriu97 @jhavukainen @aditvenk @Isalia20
